Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp

```diff
@@ -604,12 +628,20 @@ void WebSWContextManagerConnection::removeNavigationFetch(WebCore::SWServerConne
 }
 
 #if ENABLE(REMOTE_INSPECTOR) && PLATFORM(COCOA)
-void WebSWContextManagerConnection::connectToInspector(WebCore::ServiceWorkerIdentifier serviceWorkerIdentifier)
+void WebSWContextManagerConnection::connectToInspector(WebCore::ServiceWorkerIdentifier serviceWorkerIdentifier, bool isAutomaticConnection, bool immediatelyPause)
 {
     Ref channel = ServiceWorkerDebuggableFrontendChannel::create(serviceWorkerIdentifier);
     m_channels.add(serviceWorkerIdentifier, channel);
-    if (RefPtr serviceWorkerThreadProxy = SWContextManager::singleton().serviceWorkerThreadProxy(serviceWorkerIdentifier))
-        serviceWorkerThreadProxy->inspectorProxy().connectToWorker(channel);
+    if (RefPtr serviceWorkerThreadProxy = SWContextManager::singleton().serviceWorkerThreadProxy(serviceWorkerIdentifier)) {
+        Function<void()> handleFrontendInitialized = [serviceWorkerThreadProxy, serviceWorkerIdentifier] {
+            WebProcess::singleton().send(Messages::WebProcessProxy::UnpauseServiceWorkerDebuggableForInitializedInspector(serviceWorkerIdentifier));
+            serviceWorkerThreadProxy->thread().runLoop().postDebuggerTask([serviceWorkerThreadProxy](ScriptExecutionContext&) {
+                serviceWorkerThreadProxy->thread().stopRunningDebuggerTasks();
+            });
+        };
```

Comment:
👍 Looks a bit nicer now that you moved this logic here from `ServiceWorkerInspectorProxy::connectToWorker`. No more need to dig up a debuggable, which now lives in the UI process anyway.